caom2-tap/src/main/java/ca/nrc/cadc/caom2ops/CaomArtifactResolver.java

```diff
@@ -167,32 +172,57 @@ public void setAuthMethod(AuthMethod authMethod) {
     /**
      * Set RUNID value to optionally append to URLs. The runid is appended if the
      * StorageResolver implements the Traceable interface.
-     * 
-     * @param runID 
+     *
+     * @param runID
      */
     public void setRunID(String runID) {
         this.runID = runID;
     }
 
+    private void setStorageResolverAuthMethod(final StorageResolver storageResolver) {
+        try {
+            final Class<?> clazz = storageResolver.getClass();
+            final Method m = clazz.getMethod("setAuthMethod", AuthMethod.class);
+            m.invoke(storageResolver, this.authMethod);
+        } catch (NoSuchMethodException nsme) {
+            // Do nothing.  If it's missing, that's fine.
+        } catch (InvocationTargetException | IllegalAccessException classException) {
+            throw new RuntimeException(String.format("Unable to set the auth method in class %s.  Ensure the class " +
+                                                         "contains a public 'setAuthMethod(AuthMethod)' method.",
```

Comment:
I would remove the part of the message starting with "Ensure..." because we don't want the user to think they can do something about this error.